### ISSUE TITLE
"xxx is registered more than once under the same environment" error in version 1.1.0

### ISSUE THREAD

**Issue by proninyaroslav** [Milad-Akarie/injectable#177]
After upgrading from `1.0.7` to `1.1.0`, `build_runner` error occurs:
```
Injectable Generator ---------------------------------------------------
SharedPreferences [SharedPreferences] env: [] 
is registered more than once under the same environment
------------------------------------------------------------------------ 
```
I'm using the following code:
```Dart
@Singleton(as: AppSettings, dependsOn: [SharedPreferences])
class AppSettingsImpl implements AppSettings {
  final SharedPreferences pref;
  ...
}

@module
abstract class SharedPreferencesModule {
  @Singleton(env: [Env.prod, Env.dev])
  Future<SharedPreferences> get pref async => SharedPreferences.getInstance();

  @Singleton(env: [Env.test])
  Future<SharedPreferences> get testPref async => TestSharedPreferences();
}
```
Version 1.0.7 (that works) generates the following code: 
```Dart
const _test = 'test';
const _prod = 'prod';
const _dev = 'dev';
...
final sharedPreferencesModule = _$SharedPreferencesModule();
...
gh.singletonAsync<SharedPreferences>(() => sharedPreferencesModule.testPref,
    registerFor: {_test});
gh.singletonAsync<SharedPreferences>(() => sharedPreferencesModule.pref,
    registerFor: {_prod, _dev});
gh.singletonWithDependencies<AppSettings>(
    () => AppSettingsImpl(get<SharedPreferences>()),
    dependsOn: [SharedPreferences]);
```

**Comment by Milad-Akarie** [comment]
must've broken something during refactoring, I'll take a look.

**Comment by Milad-Akarie** [comment]
@proninyaroslav  fixed in `injectable_generator: ^1.1.1` please confirm and close. 
Thanks for reporting.

**Comment by proninyaroslav** [comment]
Yes, fixed. Thank you.
